sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -374,6 +375,17 @@ case class InsertIntoTable(
   }
 }
 
+case class InsertIntoDir(
```

Comment:
You are moving quite a bit of parser code to Catalyst, it also adds alot of Hive parlance to Catalyst. Maybe we should just move this into SQL/core?